TLM/TLM/UI/SubTools/LaneArrowTool.cs

```diff
@@ -108,13 +109,65 @@ public LaneArrowTool(TrafficManagerTool mainTool)
             cursorInSecondaryPanel_ = windowRect3.Contains(Event.current.mousePosition);
         }
 
+
+        private bool IsHoverValid() => IsSegmentEndHoverable(HoveredSegmentId, HoveredNodeId);
+        private bool IsSegmentEndHoverable(ushort segmentId, ushort nodeId) {
+            if(nodeId == 0 || segmentId == 0) {
+                return false;
+            }
+#if DEBUG
+            if(!Constants.ServiceFactory.NetService.IsNodeValid(nodeId) ||
+               !Constants.ServiceFactory.NetService.IsSegmentValid(segmentId)) {
+                Debug.LogError("Invalid node or segment ID");
+            }
+#endif
+            ExtSegmentEndManager segEndMan = ExtSegmentEndManager.Instance;
+            ExtSegmentEnd segEnd = segEndMan.ExtSegmentEnds[segEndMan.GetIndex(segmentId, nodeId)];
+            NetNode[] nodesBuffer = Singleton<NetManager>.instance.m_nodes.m_buffer;
+            bool IsJunction(ushort nodeId) => (nodesBuffer[nodeId].m_flags & NetNode.Flags.Junction) != 0;
```

Comment:
Why have you created local function if you can calculate bool value straight ahead??
Why argument name of your local function is overlapping with method argument name?

Reply:
Forgot to remove IsJunction. Ill do it now.